packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3574,8 +3578,12 @@ export class TransactionController extends BaseController<
         validateTxParams(transactionMeta.txParams);
       }
 
-      updatedTransactionParams =
-        this.#checkIfTransactionParamsUpdated(transactionMeta);
+      if (this.#isSimulationEnabled()) {
```

Comment:
Ah; I was wondering why this would be an adequate replacement for a general "transaction params updated" check. But I see that this `updatedTransactionParams` condition was only used to trigger resimulation. Nice that the purpose is more clear now.